source/GameData.cpp

```diff
@@ -92,6 +93,42 @@ namespace {
 
 	const Government *playerGovernment = nullptr;
 	map<const System *, map<string, int>> purchases;
+
+	// Load plugin icons and description for preferences dialog.
+	bool LoadAboutPlugin(const string &path) {
+		// Get the name of the folder containing the plugin.
+		size_t pos = path.rfind('/', path.length() - 2) + 1;
+		string name = path.substr(pos, path.length() - 1 - pos);
+
+		// Load the about text and the icon, if any.
+		plugins[name] = Files::Read(path + "about.txt");
+
+		// Create an image set for the plugin icon.
+		auto icon = make_shared<ImageSet>(name);
+
+		// Try adding all the possible icon variants.
+		if(Files::Exists(path + "icon.png"))
+			icon->Add(path + "icon.png");
+		else if(Files::Exists(path + "icon.jpg"))
+			icon->Add(path + "icon.jpg");
+
+		if(Files::Exists(path + "icon@2x.png"))
+			icon->Add(path + "icon@2x.png");
+		else if(Files::Exists(path + "icon@2x.jpg"))
+			icon->Add(path + "icon@2x.jpg");
+
+		icon->ValidateFrames();
+		spriteQueue.Add(icon);
+
+		// Enabling plugins is the default preference.
+		if(Plugins::IsEnabled(name))
+		{
+			Plugins::Set(name, true);
```

Comment:
If the plugin is already enabled why do we need to enable it again?

Reply:
Upon first launch it will not have an entry in the settings map.  Not being in the map is considered enabled.

Reply:
I can check if an entry exists before setting it but I think this would be more wasteful than always assuming because `map.find` will be called twice.  Not a big deal since the list is small.